src-docs/src/views/selection_controls/selection_controls_example.js

```diff
@@ -0,0 +1,329 @@
+          code: checkboxSource,
+        },
+      ],
+      text: (
+        <Fragment>
+          <p>
+            This component renders a basic HTML{' '}
+            <EuiCode>{'<input type="checkbox">'}</EuiCode> element. Use
+            checkboxes to allow users to select multiple options from a list.
+          </p>
+          <p>
+            Use the <EuiCode>checked</EuiCode> prop to handle the checked and
+            unchecked state. You can also use the{' '}
+            <EuiCode>indeterminate</EuiCode> prop to set an indeterminate state.
+            This state is commonly used in hierarchical checkboxes to indicate
+            that only some of the checkbox&apos;s descendants are checked.
+          </p>
+          <p>
+            Make sure to pass a label to ensure a larger clickable area and
+            ensure that screen readers will read out the label when the user is
+            focused on the input. To learn more about labels usage, go to the{' '}
+            <Link to="/forms/selection-controls/guidelines">
+              guidelines tab
+            </Link>
+            .
+          </p>
+        </Fragment>
+      ),
+      snippet: checkboxSnippet,
+      props: {
+        EuiCheckbox,
+      },
+      demo: <Checkbox />,
+      playground: CheckboxConfig,
+    },
+    {
+      title: 'Checkbox group',
+      source: [
+        {
+          type: GuideSectionTypes.JS,
+          code: checkboxGroupSource,
+        },
+      ],
+      text: (
+        <Fragment>
+          <p>
+            Use a <strong>EuiCheckboxGroup</strong> when you want to generate a
+            group of checkboxes by passing an array of objects with an ID and
+            label for each object. Use the <EuiCode>idToSelectedMap</EuiCode> to
+            indicate the IDs of the selected items.
+          </p>
+          <p>
+            When the individual labels for each radio do not provide a
+            sufficient description, pass a legend to the group.
+          </p>
+          <p>
+            Use the <EuiCode>compressed</EuiCode> prop to tighten up the spacing
+            between checkbox rows.
+          </p>
+        </Fragment>
+      ),
+      props: {
+        EuiCheckboxGroup,
+      },
+      demo: <CheckboxGroup />,
+      snippet: `<EuiCheckboxGroup
+  options={[
+    {
+      id: id1,
+      label: 'Option one',
+    },
+  ]}
+  idToSelectedMap={{ id1: true }}
+  onChange={onChange}
+/>`,
+    },
+    {
+      title: 'Radio',
+      source: [
+        {
+          type: GuideSectionTypes.JS,
+          code: radioSource,
+        },
+      ],
+      text: (
+        <Fragment>
+          <p>
+            This component renders a basic HTML{' '}
+            <EuiCode>{'<input type="radio">'}</EuiCode> element. Use radio
+            buttons to allow users to choose one option out of a list. They are
+            ideal for a list of more than 2 options, and usually no more than
+            6-8 options.
```

Comment:
Usually when you are providing a maximum count of things, you should stick to just one number. I personally think 8 is pushing it and at that point it should probably be a `<select>`. So my preference would be:
```suggestion
            6 options.
```